crates/doc/src/builder.rs

```diff
@@ -445,7 +445,15 @@ impl DocBuilder {
                 }
             } else {
                 let name = path.iter().last().unwrap().to_string_lossy();
-                let readme_path = Path::new("/").join(&path).display().to_string();
+                let readme_path = Path::new("/")
+                    .join(&path)
+                    .display()
+                    .to_string()
+                    .trim_start_matches('/')
+                    .to_owned();
+                let slash_count = readme_path.chars().filter(|&c| c == '/').count();
+                let prefix = "../".repeat(slash_count);
+                let readme_path = format!("{prefix}{readme_path}/index.html");
```

Comment:
tbh, unclear what this does.

Reply:
@mattsse Hey I optimize the code in the new commit. The code here transform the path into a relative path which can be used in a web server.

```rust
                // Generate a relative path to "index.html" based on the current directory depth and
                let name = path.iter().last().unwrap().to_string_lossy();
                let depth = path.components().count();
                let mut prefix = PathBuf::new();
                for _ in 0..depth {
                    prefix.push("..");
                }
                let readme_path = prefix.join(&path).join("index.html");
                let readme_path_str = readme_path.to_string_lossy();
                readme.write_link_list_item(&name, &readme_path_str, 0)?;
                self.write_summary_section(summary, &files, Some(&path), depth + 1)?;

```